crud/src/main/java/org/openmbee/mms/crud/services/DefaultBranchService.java

```diff
@@ -181,6 +196,55 @@ public RefJson createBranch(String projectId, RefJson branch) {
             throw new InternalErrorException(e);
         }
     }
+    public RefJson createBranchfromCommit(String projectId, RefJson parentCommitIdRef, NodeService nodeService) {
+        Instant now = Instant.now();
+
+        if(parentCommitIdRef.getParentCommitId().isEmpty()){
+            throw new BadRequestException(new RefsResponse().addMessage("parentCommitId not provided " + now.toString()));
+        }
+
+        ContextHolder.setContext(projectId);
+        Optional<Commit> parentCommit = commitRepository.findByCommitId(parentCommitIdRef.getParentCommitId());
+
+        // Get Commit object
+        String parentCommitID = parentCommit.map(Commit::getCommitId).orElseThrow(() ->
+            new BadRequestException(new RefsResponse().addMessage("parentCommitId not found " + now.toString())));
+
+        // Get Commit parentRef, the branch will be created from this
+        String parentRef = parentCommit.map(Commit::getBranchId).orElseThrow(() ->
+            new BadRequestException(new RefsResponse().addMessage("Ref from parentCommitId not found"  + now.toString())));
+
+        parentCommitIdRef.setParentRefId(parentRef);
+        ContextHolder.setContext(projectId, parentRef);
+
+        RefJson branchFromCommit = this.createBranch(projectId, parentCommitIdRef);
+
+        ContextHolder.setContext(projectId, branchFromCommit.getRefId());
+
+        // Get current nodes from database
+        List<Node> nodes = nodeRepository.findAll();
+        // Get elements from index
+        Collection<ElementJson> result = nodeGetHelper.processGetJsonFromNodes(nodes, parentCommitID, nodeService)
+            .getActiveElementMap().values();
+
+        Map<String, ElementJson> nodeCommitData = new HashMap<>();
+        for (ElementJson element : result) {
+            nodeCommitData.put(element.getId(), element);
+        }
+
+        // Update database table to match index
+        for (Node node : nodes) {
+            if(nodeCommitData.containsKey(node.getNodeId())){
+                node.setDocId(nodeCommitData.get(node.getNodeId()).getDocId());
+                node.setDeleted(false);
```

Comment:
also set node lastCommitId to be the element json's commitId

Reply:
need to do what line 189 is doing also - and not do line 189 if it wasn't branch from latest